### PR TITLE
fix(ts-typings): allow renderCallback to pass props via args

### LINKED ISSUE #1045: ts-definitions: add optional props arguments to `renderCallback`
we can use renderCallback args for props access. current TS definitions are missing this nice feature.

### PR DESCRIPTION
**Pull request by Hotell**
Closes #1045

### CODE REVIEW

File: src/ts-typings/api.d.ts [hunk 1]
```diff
@@ -37,7 +37,8 @@ export class Component<Props> extends HTMLElement {
 
   // SkateJS life cycle
   updatedCallback(previousProps: { [nameOrSymbol: string]: any }): boolean | void;
-  renderCallback(): VDOMElement<any> | VDOMElement<any>[] | null;
+  // NOTE: infering generics work only on instances, not on implementation type. So this will not give you type safety, you still have to manually annotate those props in your code
```
Comment (treshugart) [inline src/ts-typings/api.d.ts]: Feels like a given?

Comment (Hotell) [inline src/ts-typings/api.d.ts]: what do you mean by that?

Comment (treshugart) [inline src/ts-typings/api.d.ts]: It just feels like that comment explains what one would already know?

Comment (Hotell) [inline src/ts-typings/api.d.ts]: well not everyone. I was struggling with it for a while. Then I re read TS generic docs and made me tick

**Pull request merged**